src/Processors/Transforms/FilterTransform.cpp

```diff
@@ -271,5 +258,47 @@ void FilterTransform::doTransform(Chunk & chunk)
     chunk.setColumns(std::move(columns), num_filtered_rows);
 }
 
+void FilterTransform::writeIntoQueryConditionCache(MarkRangesInfoPtr mark_info)
+{
+    if (!query_condition_cache)
+        return;
+
+    if (!mark_info)
+    {
+        if (!matching_mark_info)
+            return;
+
+        query_condition_cache->write(
+            matching_mark_info->table_uuid,
+            matching_mark_info->part_name,
+            *condition_hash,
+            matching_mark_info->mark_ranges,
+            matching_mark_info->marks_count,
+            matching_mark_info->has_final_mark);
+
+        matching_mark_info = nullptr;
+        return;
+    }
+
+    if (!matching_mark_info)
+        matching_mark_info = std::static_pointer_cast<MarkRangesInfo>(mark_info->clone());
+    else
+    {
+        if (matching_mark_info->table_uuid != mark_info->table_uuid || matching_mark_info->part_name != mark_info->part_name)
```

Comment:
it is confusing to see smth named `matching` that might ATST have non-matching table id or part name. pls add some comments to clarify this code.